src/pages/workspace/workflows/WorkspaceWorkflowsApproverPage.tsx

```diff
@@ -0,0 +1,201 @@
+                text: formatPhoneNumber(PersonalDetailsUtils.getDisplayNameOrDefault(details)),
+                alternateText: formatPhoneNumber(details?.login ?? ''),
+                rightElement: roleBadge,
+                icons: [
+                    {
+                        source: UserUtils.getAvatar(details.avatar, accountID),
+                        name: formatPhoneNumber(details?.login ?? ''),
+                        type: CONST.ICON_TYPE_AVATAR,
+                        id: accountID,
+                    },
+                ],
+                errors: policyMember.errors,
+                pendingAction: policyMember.pendingAction,
+            };
+
+            if (policy?.approver === details.login) {
+                approverDetails.push(formattedMember);
+            } else {
+                policyMemberDetails.push(formattedMember);
+            }
+        });
+        return [policyMemberDetails, approverDetails];
+    }, [personalDetails, policyMembers, translate, policy?.approver, StyleUtils, isDeletedPolicyMember, policy?.owner, styles]);
+
+    const sections: MembersSection[] = useMemo(() => {
+        const sectionsArray: MembersSection[] = [];
+
+        if (searchTerm !== '') {
+            const filteredOptions = [...formattedApprover, ...formattedPolicyMembers].filter((option) => {
+                const parsedPhoneNumber = parsePhoneNumber(LoginUtils.appendCountryCode(Str.removeSMSDomain(searchTerm)));
+                const searchValue = parsedPhoneNumber.possible ? parsedPhoneNumber.number?.e164 ?? '' : searchTerm.toLowerCase();
+
+                const isPartOfSearchTerm = !!option.text?.toLowerCase().includes(searchValue) || !!option.login?.toLowerCase().includes(searchValue);
+                return isPartOfSearchTerm;
+            });
+            return [
+                {
+                    title: undefined,
+                    data: filteredOptions,
+                    shouldShow: true,
+                },
+            ];
+        }
+
+        sectionsArray.push({
+            title: undefined,
+            data: formattedApprover,
+            shouldShow: formattedApprover.length > 0,
+            indexOffset: 0,
+        });
+
+        sectionsArray.push({
+            title: translate('common.all'),
+            data: formattedPolicyMembers,
+            shouldShow: true,
+            indexOffset: formattedApprover.length,
+        });
+
+        return sectionsArray;
+    }, [formattedPolicyMembers, formattedApprover, searchTerm, translate]);
+
+    const headerMessage = useMemo(() => (searchTerm && !sections[0].data.length ? translate('common.noResultsFound') : ''), [translate, sections]);
+
+    const setPolicyApprover = (member: MemberOption) => {
+        if (!policy || !policy.approvalMode || !personalDetails?.[member.accountID]?.login) {
+            return;
+        }
+        const approver: string = personalDetails?.[member.accountID]?.login ?? policy.approver ?? policy.owner;
```

Comment:
Hm... Do we want this fallback? IMO, if `personalDetails?.[member.accountID]` is empty, we should just not let the user submit.

Reply:
On main, when approver toggle is enabled, policy owner is set as the default approver. So I don't think we should change that